internal/controller/snapshot/snapshot_adapter.go

```diff
@@ -279,62 +278,68 @@ func (a *Adapter) EnsureIntegrationPipelineRunsExist() (controller.OperationResu
 // EnsureGlobalCandidateImageUpdated is an operation that ensure the ContainerImage in the Global Candidate List
 // being updated when the Snapshot is created
 func (a *Adapter) EnsureGlobalCandidateImageUpdated() (controller.OperationResult, error) {
-	if a.snapshot.GetLabels()[gitops.SnapshotTypeLabel] != gitops.SnapshotComponentType || !gitops.IsSnapshotCreatedByPACPushEvent(a.snapshot) {
-		a.logger.Info("The Snapshot wasn't created for a single component push event, not updating the global candidate list.")
+	if (a.snapshot.GetLabels()[gitops.SnapshotTypeLabel] != gitops.SnapshotComponentType ||
+		!gitops.IsSnapshotCreatedByPACPushEvent(a.snapshot)) && !gitops.IsOverrideSnapshot(a.snapshot) {
+		a.logger.Info("The Snapshot was neither created for a single component push event nor override type, not updating the global candidate list.")
 		return controller.ContinueProcessing()
 	}
-	if gitops.IsSnapshotMarkedAsAddedToGlobalCandidateList(a.snapshot) {
-		a.logger.Info("The Snapshot's component was previously added to the global candidate list, skipping adding it.")
+
+	if gitops.IsOverrideSnapshot(a.snapshot) && !gitops.HaveAppStudioTestsSucceeded(a.snapshot) {
```

Comment:
After giving this some thought, I think we should promote the override Snapshot component images to GCL immediately and not wait to check if it passes tests. This should keep commonality with the way that the component Snapshots are handled.

Reply:
Removed the integration test check, but [STONEINTG-83](https://issues.redhat.com/browse/STONEINTG-83) mentions:
```
Allow users to create manual ‘override’ Snapshots
    This would provide a mitigation mechanism for cases when the entire application is blocked from releasing because of a single bad component build
```
I felt it is to make the entire application pass integration test.

Reply:
Yup I see your point but I think this will prevent confusion on the user's side because now both types of Snapshot will be treated the same.
We can  change this in the future based on the user's feedback, but I think this is the way to go for now.
Thanks for making the update!

Reply:
Agree. Thanks.